packages/repository/src/relations/has-many/has-many-through.helpers.ts

```diff
@@ -138,24 +139,33 @@ export function createThroughConstraint<Through extends Entity, ForeignKeyType>(
  *    keyTo: 'productId',
  *  },
  * };
+ * createThroughConstraint(resolvedMetadata, {3, name: 'a product'});
+ *
+ * >>> {productId: 3}
  * createThroughConstraint(resolvedMetadata, {id: 3, name: 'a product'});
  *
- * >>> {productId: 1}
+ * >>> {productId: 3}
  *
  * createThroughConstraint(resolvedMetadata, {id: {inq:[3,4]}});
  *
  * >>> {productId: {inq:[3,4]}}
  */
 export function createThroughFkConstraint<Target, Through extends Entity>(
   relationMeta: HasManyThroughResolvedDefinition,
-  targetInstance: Target,
+  targetInstance: Partial<Target> | undefined,
```

Comment:
I am concerned that supporting both the target instance and the target primary key is brittle. What if the target key is already an object (e.g. `ObjectID` instance or a composite key consisting of multiple SQL database columns), how are we going to tell whether the value is a key or a full model instance?

Let's rework this function to accept the key only, in the same way as `createThroughConstraint` already does.

```suggestion
  fkValue: ForeignKeyType,
```